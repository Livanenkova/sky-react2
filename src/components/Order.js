import React from 'react'
import OrderDetail from './OrderDetail'

export default class Order extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      amount: 0,
      quantity: 0,
      details: [
        {
          id: 1,
          productName: 'Война и мир - Л.Н.Толстой',
          price: 800,
          quantity: 0,
        },
        {
          id: 2,
          productName: 'Две жизни - К.Е.Антарова',
          price: 700,
          quantity: 0,
        },
        {
          id: 3,
          productName: 'Разговор с богом - Н.Д.Уолша',
          price: 1000,
          quantity: 0,
        },
        {
          id: 4,
          productName: 'Хохот Шамана - В.П.Серкин',
          price: 600,
          quantity: 0,
        },
        {
          id: 5,
          productName: 'Хроники Ехо - Макс Фрай',
          price: 400,
          quantity: 0,
        },
      ],
    }
    this.DecrementQuantityWithPrice = this.DecrementQuantityWithPrice.bind(this);
    this.IncrementQuantityWithPrice = this.IncrementQuantityWithPrice.bind(this);
  }

  IncrementQuantityWithPrice(e, index,price) {

    console.log(index)
    console.log(index)
    e.preventDefault() 
    this.setState(state => ({quantity: state.quantity + 1}))
    this.setState(state => ({amount: state.amount + price }))

    
    const data = this.state.details.map((details) => {
      console.log(data)
      if (details.id === index) {
            // this.setState(state => ({details.quantity: state.details.quantity + 1}))
        return {
          id: details.id,
          productName: details.productName,
          price: details.price,
          quantity: details.quantity + 1,
        }
      }
      return {
        id: details.id,
        productName: details.productName,
        price: details.price,
        quantity: details.quantity,
      }
    })
  }

  DecrementQuantityWithPrice(e, index,price) {
    console.log(index)
    e.preventDefault()
    // this.setState((prevState) => ({ quantity: prevState.state.quantity - 1 }))
    // this.setState(state => ({quantity: state.quantity -1}))
    // this.setState(state => ({quantity: state.amount - price }))
    this.setState((prevState) => ({ quantity: prevState.quantity - 1 }))
    this.setState((prevState) => ({ amount: prevState.amount - price }))
  }

  render() {
    const getOrderBook = this.state.details.map((details) => (
      <OrderDetail
        key={details.id}
        id={details.id}
        productName={details.productName}
        price={details.price}
        quantity={details.quantity}
        increment={this.IncrementQuantityWithPrice}
        decrement={this.DecrementQuantityWithPrice}
      />
    ))
    return (
      <div className="order">
        <div>{getOrderBook}</div>
        <div className="clear" />
        <p className="total">
          Total Quantity : <b> {`${this.state.amount}`}</b>
        </p>
        <p className="total">
          Total Price : <b> /- {`${this.state.quantity}`}</b>
        </p>
      </div>
    )
  }
}
