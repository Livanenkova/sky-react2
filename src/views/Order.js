import React from 'react'
import { Link} from 'react-router-dom'
import OrderDetail from '../components/OrderDetail'


export default class Order extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      amount: 0,
      quantity: 0,
      details: [
        {
          id: 1,
          productName: 'Война и мир - Л.Н.Толстой',
          price: 800,
          quantity: 0,
          description:'«Война́ и мир» (рус. дореф. «Война и миръ») — роман-эпопея Льва Николаевича Толстого, описывающий русское общество в эпоху войн против Наполеона в 1805—1812 годах. Эпилог романа доводит повествование до 1820 года.',
        },
        {
          id: 2,
          productName: 'Две жизни - К.Е.Антарова',
          price: 700,
          quantity: 0,
          description:'В книге Конкордии Евгеньевны Антаровой "Две жизни" впервые в беллетристической форме даются яркие и глубокие образы Великих Учителей, выписанные с огромной любовью, показан их самоотверженный труд на благо человечества. Учение, изложенное в книгах "Живой Этики", как бы проиллюстрировано судьбами героев книги "Две жизни".',
        },
        {
          id: 3,
          productName: 'Разговор с богом - Н.Д.Уолша',
          price: 1000,
          quantity: 0,
          description:'Перед читателем - необычный документ нашего времени: послание от Бога - своеобразная программа духовной революции, исчерпывающая все сферы познания и деятельности человека - от сугубо личной до планетарной.',
        },
        {
          id: 4,
          productName: 'Хохот Шамана - В.П.Серкин',
          price: 600,
          quantity: 0,
          description:'В 1997 году психолог Владимир Серкин случайно познакомился с человеком, которого все окружающие считали шаманом.',
        },
        {
          id: 5,
          productName: 'Хроники Ехо - Макс Фрай',
          price: 400,
          quantity: 0,
          description:'Серия книг от одного из самого известного автора современности Макса Фрая, истории про Вселенную Ехо',
        },
      ],
    }
    this.DecrementQuantityWithPrice = this.DecrementQuantityWithPrice.bind(this);
    this.IncrementQuantityWithPrice = this.IncrementQuantityWithPrice.bind(this);
  }
  

  IncrementQuantityWithPrice(e, index,price) {
    e.preventDefault() 
    this.setState(state => ({quantity: state.quantity + 1}))
    this.setState(state => ({amount: state.amount + price }))

  }

  DecrementQuantityWithPrice(e, index,price) {
    e.preventDefault()
    this.setState((prevState) => ({ quantity: prevState.quantity - 1 }))
    this.setState((prevState) => ({ amount: prevState.amount - price }))
  }

  render() {
    const getOrderBook = this.state.details.map((details) => (
      <OrderDetail
        key={details.id}
        id={details.id}
        description={details.description}
        productName={details.productName}
        price={details.price}
        quantity={details.quantity}
        increment={this.IncrementQuantityWithPrice}
        decrement={this.DecrementQuantityWithPrice}
      />
    ))
    return (
      <div className="order">
        <div>{getOrderBook}</div>
        <div className="clear" />
        <p className="total">
          Total Quantity : <b> {`${this.state.amount}`}</b>
        </p>
        <p className="total">
          Total Price : <b> /- {`${this.state.quantity}`}</b>
        </p>
        <Link to="/user-order" state={this.state}>User-Order</Link>
      </div>
    )
  }
}
