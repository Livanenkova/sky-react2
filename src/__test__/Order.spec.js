/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer';
import {Router } from 'react-router-dom';
import Order from '../views/Order';

const { createMemoryHistory } = require("history");


it('отображается корректно', () => {
  const history = createMemoryHistory();
  const tree = renderer.create(<Router location={history.location} navigator={history}><Order/></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
