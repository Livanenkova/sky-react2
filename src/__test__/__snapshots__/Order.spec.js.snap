// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`отображается корректно 1`] = `
<div
  className="order"
>
  <div />
  <div
    className="clear"
  />
  <p
    className="total"
  >
    Total Quantity : 
    <b>
       
      0
    </b>
  </p>
  <p
    className="total"
  >
    Total Price : 
    <b>
       /- 
      0
    </b>
  </p>
  <a
    href="/user-order"
    onClick={[Function]}
  >
    User-Order
  </a>
</div>
`;
